cmd/utils.go

```diff
@@ -546,3 +560,128 @@ func generateRGWPortToUse() string {
 	}
 	return "notfound"
 }
+
+// GetFileType checks wether a specified data is directory, a block device or something else
+// function borrowed from https://github.com/andrewsykim/kubernetes/blob/2deb7af9b248a7ddc00e61fcd08aa9ea8d2d09cc/pkg/util/mount/mount_linux.go#L416
+func GetFileType(pathname string) (string, error) {
+	finfo, err := os.Stat(pathname)
+	if os.IsNotExist(err) {
+		return "notfound", fmt.Errorf("path %q does not exist", pathname)
+	}
+	// err in call to os.Stat
+	if err != nil {
+		return "error", err
+	}
+
+	mode := finfo.Sys().(*syscall.Stat_t).Mode
+	switch mode & syscall.S_IFMT {
+	case syscall.S_IFSOCK:
+		return "socket", nil
+	case syscall.S_IFBLK:
+		return "blockdev", nil
+	case syscall.S_IFCHR:
+		return "chardev", nil
+	case syscall.S_IFDIR:
+		return "directory", nil
+	case syscall.S_IFREG:
+		return "file", nil
+	}
+
+	return "error", fmt.Errorf("only recognize file, directory, socket, block device and character device")
+}
+
+// TestBinaryExist tests if a binary is present on the system
+func TestBinaryExist(binary string) bool {
+	binary, err := exec.LookPath(binary)
+	if err != nil {
+		log.Fatal(binary + " is not installed!")
+	}
+	return true
+}
+
+// GetDiskFormat returns information about a disk such as filesystem and partition table
+func GetDiskFormat(disk string) string {
+	TestBinaryExist("blkid")
+
+	out, err := exec.Command("blkid", "-p", "-s", "TYPE", "-s", "PTTYPE", "-o", "export", disk).Output()
+	if err != nil {
+		// Disk device does not have any label.
+		// For `blkid`, if the specified token (TYPE/PTTYPE, etc) was
+		// not found, or no (specified) devices could be identified, an
+		// exit code of 2 is returned.
+		// We are not 100% that this is the problem, but the code called before this already made sure
+		// that the device was a block special device, so if blkid fails that's probably because there is
+		// nothing to see. My first assumption is that the device does not have a partition label
+		log.Fatal("The disk" + disk + " has no partition label, please create it." +
+			"\nAlso if the disk was an OSD you need to zap it (e.g: with 'ceph-disk zap')")
+	}
+	return string(out)
+}
+
+// GetDiskPartitions returns the list of partitions on a disk
+func GetDiskPartitions(disk string) []string {
+	TestBinaryExist("parted")
+
+	out, err := exec.Command("parted", "-s", "-m", disk, "print").Output()
+	if err != nil {
+		log.Fatal(err)
+	}
+	var partitions []string
+	lines := strings.Split(string(out), "\n")
+	// build a slice
+	for _, l := range lines {
+		if len(l) <= 0 {
+			// Ignore empty line.
+			continue
+		}
+		partitions = append(partitions, l)
+	}
+	return partitions
+}
+
+// DeviceOpened checks if block device in use by calling Open with O_EXCL flag.
+func DeviceOpened(pathname string) (bool, error) {
+	return ExclusiveOpenFailsOnDevice(pathname)
```

Comment:
What is the purpose of this empy function ? Can't we call this function directly ?

Reply:
Correct